tensorflow/python/ops/image_ops_impl.py

```diff
@@ -1724,8 +1724,22 @@ def convert_image_dtype(image, dtype, saturate=False, name=None):
 
   Returns:
     `image`, converted to `dtype`.
+  
+  Usage Example:
+    ```python
+    >> import tensorflow as tf
+    >> x = tf.random.normal(shape=(256, 256, 3), dtype=tf.float32)
+    >> tf.image.convert_image_dtype(x, dtype=tf.float16, saturate=False)
+    ```
+  
+  Raises:
+    AttributeError: Type object has no attribute is_integer or is_floating
   """
   image = ops.convert_to_tensor(image, name='image')
+  if hasattr(dtype, 'is_floating') or hasattr(dtype, 'is_integer'):
```

Comment:
Can you rewrite this to not use hasattr?

Reply:
Sure. Any suggestions?

Reply:
if not dtype.is_floating and not dtype.is_integer: raise ?

Reply:
This looks nice. But wouldn't it throw an error at the `if` condition when is_floating attribute does not exists in the dtype.
As an example:

```python
dtype = int
if not dt.is_floating and not dt.is_integer:
  raise AttributeError("Type object has no attribute is_integer or is_floating")
```

This will raise the error

```python
AttributeError                            Traceback (most recent call last)
<ipython-input-10-4228559210de> in <module>()
----> 1 if not dt.is_floating and not dt.is_integer:
      2     raise AttributeError("Type object has no attribute is_integer or is_floating")

AttributeError: type object 'int' has no attribute 'is_floating'
```